node/out_response.js

```diff
@@ -195,6 +195,15 @@ TChannelOutResponse.prototype.sendCallResponseContFrame = function sendCallRespo
 
 TChannelOutResponse.prototype.sendError = function sendError(codeString, message) {
     var self = this;
+
+    if (self.inreq.connection.closing) {
+        self.logger.info('ignoring outresponse.sendError on a closed connection', {
+            codeString: codeString,
+            errorMessage: message
+        });
+        return;
```

Comment:
We can get here without being in the done or error state. This should also emit finish before returning.